app.js

```diff
@@ -111,24 +110,29 @@ window.onload = async () => {
     auth = firebase.auth();
 
     auth.onAuthStateChanged(async (user) => {
-      let idToken = '';
-      if (user) {
-        idToken = await user.getIdToken();
-        if (!user.isAnonymous) {
-          localforage.clear();
-          inactivityTime();
-          const firstSignInTime = new Date(user.metadata.creationTime).toISOString();
-          appState.setState({ participantData: { firstSignInTime } });
-        } 
-      }
+        let idToken = '';
+        if (user) {
+            idToken = await user.getIdToken();
+            if (!user.isAnonymous) {
+                localforage.clear();
+                const firstSignInTime = new Date(user.metadata.creationTime).toISOString();
+                appState.setState({ participantData: { firstSignInTime } }); // TODO: potential issue with firstSignInTimestamp
+            }
+        }
 
-      appState.setState({ idToken });
+        appState.setState({ idToken });
+
+        await router();
+
+        if (user && !user.isAnonymous) {
+            inactivityTime();
+        }
     });
 
     const footer = document.getElementById('footer');
-    footer.innerHTML = footerTemplate();
-    
-    router();
+    if (footer) {
```

Comment:
I see during my local tests that separated footer template creates not found errors for app version rendering.
I'll try moving footer template to index.html file.

Reply:
I moved the footer into index.js. There's still a timing issue with the service worker and app version on hard restarts (Chrome cmd + shift + r). That has been there since it was implemented. I can try to look into it more when I inspect the 'first time sign in' issue.